### PR TITLE
AWS 3 task: BE part

### PR DESCRIPTION
**Pull request by OlgaGnatenko**
AWS 3 task: BE part

Backend is available here: 
https://jh2vjbyda1.execute-api.eu-west-1.amazonaws.com/dev/products
https://jh2vjbyda1.execute-api.eu-west-1.amazonaws.com/dev/products/:id 

Frontend part of the app available here:
https://d2joi2c0vsqdd3.cloudfront.net/

**What was done**
1 - product-service serverless.yml contains configuration for 2 lambda functions
2, 3 - lambda functions getProducts and getProductById are located in two separate files
4 - Frontend application is integrated with product service (/products API) and products from product-service are represented on Frontend. 

Additional:

- Async/await is used in lambda functions (used in just one function since we are currently storing the products list as hardcoded)
- Lambda handlers (getProductsList, getProductsById) code is written not in 1 single module (file) and separated in codebase.
- Main error scenarious are handled by API ("Product not found" error, try catch blocks are used in lambda handlers).

### CODE REVIEW

**Comment by vvkotov** [comment]
Hi, will you push something today?

**Comment by OlgaGnatenko** [comment]
> Hi, will you push something today?

Just did. If possible, could you please have a look? :)

**Review by vvkotov — APPROVED** [review]

File: product-service/productById.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+"use strict";
+const products = require("./products");
+
+module.exports.getProductById = async (event) => {
+  try {
+    let id = null;
```
Comment (kaseopea) [inline product-service/productById.js:6]: no need to initialize id variable

you can use es6 destructuring - `const { id } = event.pathParameters;`
or simply  `const id = event.pathParameters['id'];`

File: product-service/productById.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+module.exports.getProductById = async (event) => {
+  try {
+    let id = null; 
+    if (event.pathParameters) {
+      id =  event.pathParameters.id;
+    }
+    if (!id)
+      return {
+        headers: {
+          "Access-Control-Allow-Origin": "*",
```
Comment (kaseopea) [inline product-service/productById.js:13]: According to DRY principle would be nice to store headers in a separate variable as long as they repeat several times.

File: product-service/productById.js (hunk 1 above)
Comment (OlgaGnatenko) [reply product-service/productById.js:6]: It seems that when there is no pathParameter, the property pathParameters does not exist (I had that issue with queryParameters), so this is added to handle such situation and avoid 500 error.

**Pull request merged**